src/content/services/lookup.js

```diff
@@ -8,7 +8,7 @@ function load(word, lang, targetLang) {
     .then(data => ({ lang: lang, data: data }));
 }
 
-export default function lookup(word, targetLang, sourceLang) {
-  const lang = detectLanguage() || sourceLang || 'en';
+export default function lookup(word, targetLang, sourceLang, autoDetectLang) {
+  const lang = autoDetectLang ? detectLanguage() || sourceLang || 'en' : sourceLang || 'en';
```

Comment:
Hi Branislav,

Thanks for the contribution and sorry for the delay!

I've also seen the language detection not always working correctly. I think we should somehow ask the user to select their target language on first use, and stick to that selected option as the default language.

So for now, I'd propose to just give priority to the `sourceLang` from the options.

```suggestion
  const lang = sourceLang || detectLanguage() || 'en';
```